doc/source/tutorial/optimize.rst

```diff
@@ -1347,6 +1350,215 @@ Preconditioning is an art, science, and industry. Here, we were lucky
+"box constraints" or "simple bounds". These constraints can be applied using the bounds argument of :func:`linprog`.
+As noted in the :func:`linprog` documentation, the default value of bounds is (0, None), meaning that the
+lower bound on each decision variable is 0, and the upper bound on each decision variable is infinity:
+all the decision variables are non-negative. Our bounds are different, so we specify the lower and upper bound on each
+decision variable as a tuple and group these tuples into a list.
+
+
+Finally, we can solve the transformed problem using :func:`linprog`.
+
+::
+
+    >>> import numpy as np
+    >>> from scipy.optimize import linprog
+    >>> c = np.array([-29.0, -45.0, 0.0, 0.0]).T
+    >>> A_ub = np.array([[1.0, -1.0, -3.0, 0.0],
+    ...                [-2.0, 3.0, 7.0, -3.0]])
+    >>> b_ub = np.array([[5.0],
+    ...                [-10.0]])
+    >>> A_eq = np.array([[2.0, 8.0, 1.0, 0.0],
+    ...                 [4.0, 4.0, 0.0, 1.0]])
+    >>> b_eq = np.array([[60.0],
+    ...                [60.0]])
+    >>> x0_bounds = (0, None)
+    >>> x1_bounds = (0, 5.0)
+    >>> x2_bounds = (-np.inf, 0.5)  # +/- np.inf can be used instead of None
+    >>> x3_bounds = (-3.0, 0)
+    >>> bounds = [x0_bounds, x1_bounds, x2_bounds, x3_bounds]
+    >>> result = linprog(c, A_ub=A_ub, b_ub=b_ub, A_eq=A_eq, b_eq=b_eq, bounds=bounds)
+    >>> print(result)
+        con: array([42.58855175, 45.54023356])  # may vary
+        fun: -143.26958745795812  # may vary
+    message: 'The algorithm terminated successfully and determined that the problem is infeasible.'
+        nit: 4
+      slack: array([  2.89860769, -12.11677835])  # may vary
+     status: 2
+    success: False
+          x: array([ 2.41503507,  1.62741268, -0.43792331, -1.71002454])  # may vary
+
+The result shows our problem is infeasible. It means that there is no solution vector that satisfies all the
+constraints. What did we do wrong? If we check the code carefully, we can notice that we've set the :math:`x_3` bound
+constraint wrong. We've set the maximum bound for :math:`x_3` to :math:`0`, but it is not needed.
+
+We can solve it with correct bound settings:
+
+::
+
+    >>> x3_bounds = (-3.0, None)
+    >>> bounds = [x0_bounds, x1_bounds, x2_bounds, x3_bounds]
+    >>> result = linprog(c, A_ub=A_ub, b_ub=b_ub, A_eq=A_eq, b_eq=b_eq, bounds=bounds)
+    >>> print(result)
+         con: array([9.78840831e-09, 1.04662945e-08])  # may vary
+         fun: -505.97435889013434  # may vary
+     message: 'Optimization terminated successfully.'  # may vary
+         nit: 4
+       slack: array([ 6.52747190e-10, -2.26730279e-09])  # may vary
+      status: 0
+     success: True
+           x: array([ 9.41025641,  5.17948718, -0.25641026,  1.64102564])  # may vary
+
+The result shows the optimization was succeeded.
+We can check the objective value (``result.fun``) is same as :math:`c^Tx`:
+
+::
+
+    >>> x = np.array([result.x]).T
+    >>> print(c.T @ x)
```

Comment:
```suggestion
    >>> print(c @ x)
```
I would argue that we shouldn't show transposes. Even if it makes it look more like the math, users should know - or be taught - that it's not necessary.

Reply:
OK. I got it. Let's remove transpose.